scripts/junit2issue.py

```diff
@@ -0,0 +1,194 @@
+            generated issue content string
+    '''
+    _s = Template(template)
+    content = _s.substitute(context)
+    # logging.info(content)
+    return content
+
+def github_login():
+    '''
+        github_login
+    '''
+
+    access_token = os.getenv('GITHUB_ACCESS_TOKEN')
+
+    if not access_token:
+        print("no access token find in system environment variable\n")
+        print("please set GITHUB_ACCESS_TOKEN with:\n")
+        print("export GITHUB_ACCESS_TOKEN=<your github access token>\n")
+        return None
+
+    # using an access token
+    return Github(access_token)
+
+def check_issue_existing(repository, report):
+    '''
+        check whether the report issue exists
+        @repository: in repo target
+        @report: dict fo reporting content
+        return:
+            existing issue or None
+    '''
+    tod = datetime.datetime.now()
+    d = datetime.timedelta(days=60)
+    a = tod - d
+    open_issues = repository.get_issues(state='open', since=a, labels=['bug'])
+    for _issue in open_issues:
+        if report['CASE_NAME'] in _issue.title:
+            return _issue
+        if report['LOGS'] in _issue.body:
+            return _issue
+
+    return None
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG,
+                        format='[%(filename)s:%(lineno)d] %(threadName)s %(message)s')
+    pfile = 'lpcxpresso55s28.xml'
+    vfolder = 'zephyr-v2.6.0-286-g46029914a7ac'
+    auto_report = False
+    try:
+
+        opts, args = getopt.getopt(sys.argv, "hav:i:", ["vfolder=", "pfile="])
+        for opt, arg in opts:
+            if opt == '-h':
+                print('junit2issue.py -v <vfolder> -i <pfile>')
+                sys.exit(0)
+            elif opt in ("-i", "--pfile"):
+                pfile = arg
+            elif opt in ("-v", "--vfolder"):
+                vfolder = arg
+            elif opt == '-a':
+                auto_report = True
+        XML_FILE = os.path.join('..\\results', vfolder, pfile)
+    except getopt.GetoptError:
+        print('junit2issue.py -v <vfolder> -i <pfile>')
+        XML_FILE = os.path.join('..\\results\\zephyr-v2.6.0-286-g46029914a7ac\\',
+                                'lpcxpresso55s28.xml')
+
+    XML = JUnitXml.fromfile(XML_FILE)
+    #REPORTS = parser_testsuites_for_result(XML, Error)
+    REPORTS = parser_testsuites_for_result(XML, Failure)
+    logging.info(len(REPORTS))
+    if not auto_report:
+        print("Do you want to report issue?Y/N")
+        CHOOSE = input()
+        if CHOOSE in ['n', 'N']:
+            print("Bye\n")
+            sys.exit(0)
+
+    g = github_login()
+    my_repo = os.getenv('REPORT_REPO')
+    repo = g.get_repo(my_repo)
+    label1 = repo.get_label("area: Tests")
+    label2 = repo.get_label("bug")
+
+    for rep in REPORTS:
+        if not auto_report:
+            print("\r\ndo you want to report ", rep['CASE_NAME'])
+            CHOOSE = input()
+            if CHOOSE in ['n', 'N']:
```

Comment:
check for yes to continue, not 'no', this way you will avoid mistakes typing something else and filing the wrong issue.

Reply:
ok